cli/provider_cache.go

```diff
@@ -79,118 +79,80 @@ func InitProviderCacheServer(opts *options.TerragruntOptions) (*ProviderCacheSer
-		return err
-	}
-	return nil
-}
-
-// providerCacheEnvironment returns TF_* name/value ENVs, which we use to force terraform processes to make requests through our cache server (proxy) instead of making direct requests to the origin servers.
-func providerCacheEnvironment(opts *options.TerragruntOptions, cliConfigFile string) map[string]string {
-	envs := make(map[string]string)
-
-	for _, registryName := range opts.ProviderCacheRegistryNames {
-		envName := fmt.Sprintf(terraform.EnvNameTFTokenFmt, strings.ReplaceAll(registryName, ".", "_"))
-		// We use `TF_TOKEN_*` for authentication with our private registry (cache server).
-		// https://developer.hashicorp.com/terraform/cli/config/config-file#environment-variable-credentials
-		envs[envName] = opts.ProviderCacheToken
-	}
-
-	if !opts.ProviderCacheDisablePartialLockFile {
-		// By using `TF_PLUGIN_CACHE_MAY_BREAK_DEPENDENCY_LOCK_FILE` we force terraform to generate `.terraform.lock.hcl` only based on cached files, otherwise it downloads three files (provider zip archive, SHA256SUMS, sig) from the original registry to calculate hashes.
-		// https://developer.hashicorp.com/terraform/cli/config/config-file#allowing-the-provider-plugin-cache-to-break-the-dependency-lock-file
-		envs[terraform.EnvNameTFPluginCacheMayBreakDependencyLockFile] = "1"
-	}
-
-	// By using `TF_CLI_CONFIG_FILE` we force terraform to use our auto-generated cli configuration file.
-	// https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_cli_config_file
-	envs[terraform.EnvNameTFCLIConfigFile] = cliConfigFile
-	// Clear this `TF_PLUGIN_CACHE_DIR` value since we are using our own caching mechanism.
-	// https://developer.hashicorp.com/terraform/cli/config/environment-variables#tf_plugin_cache_dir
-	envs[terraform.EnvNameTFPluginCacheDir] = ""
+	maps.Copy(cloneOpts.Env, env)
 
-	return envs
+	return shell.RunTerraformCommandWithOutput(ctx, cloneOpts, args...)
 }
 
-// createLocalCLIConfig creates a local CLI configuration that merges the default/user configuration with our Private Registry configuration.
-// We don't want to use Terraform's `plugin_cache_dir` feature because the cache is populated by our Terragrunt Provider Cache server, and to make sure that no Terraform process ever overwrites the global cache, we clear this value.
-// In order to force Terraform to queries our registry instead of the original one, we use the section below.
+// createLocalCLIConfig creates a local CLI config that merges the default/user configuration with our Provider Cache configuration.
+// We don't want to use Terraform's `plugin_cache_dir` feature because the cache is populated by our Terragrunt Provider Cache cache, and to make sure that no Terraform process ever overwrites the global cache, we clear this value.
+// In order to force Terraform to queries our cache cache instead of the original one, we use the section below.
```

Comment:
More search/replace issues resulting in "cache cache," it seems. I won't highlight others; do a scan through, as there are prob more!

Reply:
Fixed everywhere, thanks.